### ISSUE TITLE
SIGSEGV when absolutely slamming cesium with world generation writes

### ISSUE THREAD

**Issue by ryleu** [CesiumMC/cesium-fabric#19]
I have big globes and distant horizons and when pre-generating a large radius using chunk (set DH's CPU impact to "I PAID FOR THE WHOLE CPU" or else it won't be able to keep up), Cesium triggered a segmentation fault. Here's the JVM log:

[hs_err_pid481.log](https://github.com/user-attachments/files/17451972/hs_err_pid481.log)

And here's the game log:

https://mclo.gs/wo9XZSO

I wouldn't worry too much about this crash if not for the JVM error, given that this is such an absurd situation. Contact me if you want an exported MultiMC / Prism Launcher instance with the mod setup.

**Comment by ryleu** [comment]
it's possible this was caused by fast quit. i don't plan on regenerating about 8 gb worth of chunks soon, but the next time i do, i'll see if i can reproduce it with only cesium, big globe, and fast quit

**Comment by yamayaki1** [comment]
Chances are that fastquit causes the server to close the database before it can actually commit all the changes.

While I’m not happy that the JVM crashes, that’s what I’ve observed while working on this mod in the past.

I can probably add some code to fix compatibility with fastquit.

**Comment by yamayaki1** [comment]
Okay, I can't really reproduce this with only Cesium and fastquit. I guess there's something more going on.

**Comment by ryleu** [comment]
Yeah, it probably has to do with being slammed. I'll see what minimal set of mods i can use to reproduce.

**Comment by CubeBeveled** [comment]
maybe it has alot of stuff to commit to the db and for some reason the db gets closed before its done
